spec/intl.html

```diff
@@ -117,5 +117,32 @@ <h1>Intl.getCanonicalLocales ( _locales_ )</h1>
         1. Return CreateArrayFromList(_ll_).
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-intl.supportedvaluesof">
+      <h1>Intl.supportedValuesOf ( _key_ )</h1>
+
+      <p>
+      When the `supportedValuesOf` method is called with argument _key_ , the following steps are taken:
+      </p>
+
+      <emu-alg>
+        1. Let _key_ be ? ToString(_key_).
+        1. If _key_ is *"calendar"*, then
+          1. Let _list_ be AvailableCanonicalCalendars( ).
+        1. Else if _key_ is *"collation"*, then
+          1. Let _list_ be AvailableCanonicalCollations( ).
+        1. Else if _key_ is *"currency"*, then
+          1. Let _list_ be AvailableCanonicalCurrencies( ).
+        1. Else if _key_ is *"numberingSystem"*, then
+          1. Let _list_ be AvailableCanonicalNumberingSystems( ).
+        1. Else if _key_ is *"timeZone"*, then
+          1. Let _list_ be AvailableCanonicalTimeZones( ).
+        1. Else if _key_ is *"unit"*, then
+          1. Let _list_ be AvailableCanonicalUnits( ).
+        1. Else,
+          1. Throw a *RangeError* exception.
```

Comment:
a RangeError, from an unknown key? i'd expect a TypeError here.

Reply:
Why is that? the type of the key is String, we also throw RangeError in https://tc39.es/ecma402/#sec-getoption right?

Reply:
other places in ECMA402 we also throw RangeError
https://tc39.es/ecma402/#sec-canonicalcodefordisplaynames
https://tc39.es/ecma402/#sec-canonicalizelocalelist
"a. If collation does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception."
in https://tc39.es/ecma402/#sec-initializecollator
"a. If calendar does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
a. If numberingSystem does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
b. If the result of ! [IsValidTimeZoneName](https://tc39.es/ecma402/#sec-isvalidtimezonename)(timeZone) is false, then
"
in https://tc39.es/ecma402/#sec-initializedatetimeformat

"a. If calendar does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
a. If collation does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
a. If numberingSystem does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
"
in https://tc39.es/ecma402/#sec-Intl.Locale
also
https://tc39.es/ecma402/#sec-apply-options-to-tag
https://tc39.es/ecma402/#sec-initializenumberformat

"

Reply:
That seems very strange - RangeErrors are for when something is out of a range - enums aren't a "range", conceptually or otherwise.

Reply:
[RangeError](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-native-error-types-used-in-this-standard-rangeerror) (emphasis mine)
> Indicates a value that is not in the **set** or range of allowable values.

It's also used this way in ECMA-262:
* [NumberToBigInt](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-numbertobigint)
* [String.fromCodePoint](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.fromcodepoint)
* [String.prototype.normalize](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.normalize) (an especially good analog)
* [InitializeTypedArrayFromArrayBuffer](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-initializetypedarrayfromarraybuffer)

Reply:
fromCodePoint is an actual ascii range; the AOs are ranges as well; but you’re right that normalize is a good analog. It still seems super weird to me, since an enum simply isn’t a range of any kind.